src/polymer/polymer-element-mixin.ts

```diff
@@ -34,50 +34,52 @@ export interface Options {
 
 export class ScannedPolymerElementMixin extends ScannedElementMixin implements
     ScannedPolymerExtension {
-  properties: ScannedPolymerProperty[] = [];
-  methods: ScannedMethod[] = [];
-  observers: Observer[] = [];
-  listeners: {event: string, handler: string}[] = [];
-  behaviorAssignments: ScannedBehaviorAssignment[] = [];
+  readonly properties: ScannedPolymerProperty[] = [];
+  readonly methods: ScannedMethod[] = [];
+  readonly observers: Observer[] = [];
+  readonly listeners: {event: string, handler: string}[] = [];
+  readonly behaviorAssignments: ScannedBehaviorAssignment[] = [];
   // FIXME(rictic): domModule and scriptElement aren't known at a file local
   //     level. Remove them here, they should only exist on PolymerElement.
   domModule: dom5.Node|undefined = undefined;
   scriptElement: dom5.Node|undefined = undefined;
   pseudo: boolean = false;
-  abstract: boolean = false;
+  readonly abstract: boolean = false;
+  readonly sourceRange: SourceRange;
 
-  constructor(options?: Options) {
+  constructor({
+    name,
+    jsdoc,
+    description,
+    summary,
+    privacy,
+    sourceRange,
+    mixins,
+    astNode
+  }: Options) {
```

Comment:
IMO it's a bit awkward, but for classes with tons of members I don't know of a better option that preserves type safety.